pkg/util/netops/netops.go

```diff
@@ -259,556 +243,480 @@ func (ns *NS) GetIPNet(linkName string) ([]*net.IPNet, error) {
-		if err != nil {
-			return "", fmt.Errorf("failed to obtain addresses assigned to %s (netns: %s)", name, ns.Path)
-		}
-		for _, addr := range addrs {
-			if addr.Equal(ip) {
-				names = append(names, name)
-				break
-			}
-		}
-	}
+func (l *link) SetNamespace(target Namespace) error {
 
-	if len(names) == 0 {
-		return "", fmt.Errorf("failed to find interface that has %s on netns %s", ip.String(), ns.Path)
-	}
-	if len(names) > 1 {
-		return "", fmt.Errorf("multile interfaces have %s on netns %s: %s", ip.String(), ns.Path, strings.Join(names, ", "))
+	if err := l.ns.handle.LinkSetNsFd(l.nlLink, target.(*namespace).fd()); err != nil {
+		return fmt.Errorf("failed to change network namespace of interface %s from %s to %s: %w", l.Name(), l.ns.path, target.Path(), err)
 	}
 
-	return names[0], nil
+	l.ns = target.(*namespace)
+
+	return nil
 }
 
-func (ns *NS) LinkList() ([]netlink.Link, error) {
+func (l *link) SetName(name string) error {
 
-	links, err := ns.handle.LinkList()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get a list of interfaces: %w", err)
+	if err := l.ns.handle.LinkSetName(l.nlLink, name); err != nil {
+		return fmt.Errorf("failed to change name of interface %s on %s to %s: %w", l.Name(), l.ns.path, name, err)
 	}
 
-	return links, nil
+	return nil
 }
 
-func (ns *NS) LinkByName(name string) (netlink.Link, error) {
+func (l *link) SetUp() error {
 
-	link, err := ns.handle.LinkByName(name)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find interface %s: %w", name, err)
+	if err := l.ns.handle.LinkSetUp(l.nlLink); err != nil {
+		return fmt.Errorf("failed to set link state up: %s: %w", l.Name(), err)
 	}
-
-	return link, nil
+	return nil
 }
 
-// LinkAdd creates a new link with an attribute specified by link
-func (ns *NS) LinkAdd(linkName string, link netlink.Link) error {
-	attrs := link.Attrs()
-	*attrs = netlink.NewLinkAttrs()
-	attrs.Name = linkName
-	if err := ns.handle.LinkAdd(link); err != nil {
-		return fmt.Errorf("failed to create an interface of %s: %s:  %w", link.Type(), linkName, err)
+func (l *link) Delete() error {
+
+	if err := l.ns.handle.LinkDel(l.nlLink); err != nil {
+		return fmt.Errorf("failed to delete an interface of %s: %s:  %w", l.Name(), l.Type(), err)
 	}
 	return nil
 }
 
-// LinkDel deletes a link
-func (ns *NS) LinkDel(linkName string) error {
-	link, err := ns.handle.LinkByName(linkName)
-	if err != nil {
-		return fmt.Errorf("failed to get interface %s: %w", linkName, err)
-	}
-	if err := ns.handle.LinkDel(link); err != nil {
-		return fmt.Errorf("failed to delete an interface of %s: %s:  %w", linkName, link.Type(), err)
-	}
-	return nil
+type Device interface {
+	getLink() netlink.Link
 }
 
-// LinkSetMaster sets a master device of a link
-func (ns *NS) LinkSetMaster(linkName, masterName string) error {
-	master, err := ns.handle.LinkByName(masterName)
-	if err != nil {
-		return fmt.Errorf("failed to get interface %s: %w", masterName, err)
+type VEth struct {
```

Comment:
I think it's deliberately to use `VEth` instead of `Veth`? Any reason here?

Reply:
@yoheiueda I'm not sure I'm convinced because I saw some popular packages are using `Veth`, like https://github.com/weaveworks/weave/blob/master/net/veth.go and https://github.com/vishvananda/netlink/blob/main/link.go, but I think it's not a real problem.